liquibase-standard/src/main/java/liquibase/changelog/visitor/ValidatingVisitor.java

```diff
@@ -168,6 +169,52 @@ public void visit(ChangeSet changeSet, DatabaseChangeLog databaseChangeLog, Data
         }
     } // public void visit(...)
 
+    /**
+     * MongoDB's extension was incorrectly messing with CreateIndex checksum when the extension was added to the lib folder
+     * but a database other than mongodb was used. This method checks:
+     * * is it a CreateIndex change?
+     * * are we not using mongo?
+     * * do we have mongo extension loaded?
+     * * If I use CreateIndex from mongo extension, does the checksum matches?
+     * If everything matches than we fix the checksum on the database and say it's fine to continue.
+     */
+    private boolean isItAnOldButCorrectChecksumVersionGeneratedByABuggyExtension(ChangeSet changeSet, RanChangeSet ranChangeSet, DatabaseChangeLog databaseChangeLog) {
```

Comment:
@StevenMassaro I added this new hack here to handle and fix incorrect checksums instead of just failing validation.

Reply:
Oh this is yucky. Is this a product requirement?

Reply:
Yes, this here -> https://datical.atlassian.net/browse/DAT-15258?focusedCommentId=128295

Reply:
I don't think this code should be added to the Liquibase codebase. This is hard to test, and will have edge cases that we cannot imagine now. I fear that adding this could break some part of Liquibase for the majority of users who do not fall into this very narrow window of users we're trying to fix for. Additionally, I don't want to set a precedent that we will be adding in all kinds of hacky workarounds (no offense to your fix or code) to fix problems caused by other extensions.

The proper solution to a problem like this is education and documentation in the event that someone falls into the unlikely situation of meeting all of these criteria.

Reply:
@StevenMassaro @suryaaki2 I replaced the magic code by a warning message. It is working like that:

```
Unexpected error running Liquibase: Validation Failed:
     1 changesets check sum
          changelogs/changelog/simple_changelog.xml::3::fl4 was: 9:6189b7cbca33bf5dec62fd48dd75c473 but is now: 9:1ef3d63cb4491a88e7d50778226c49ca
     1 changes have validation failures
          Checksum value 9:6189b7cbca33bf5dec62fd48dd75c473 for CreateIndexChange probably was created by an old version of mongodb extension and needs to be fixed manually, changelogs/changelog/simple_changelog.xml::3::fl4
```

Reply:
Are we able to add some more instructions about how to fix it manually to that message?